Sources/RayTracer.Library/Serialization/Serializers/SceneSerializer.cs

```diff
@@ -19,7 +20,7 @@ public override void Serialize(Utf8JsonWriter writer, Scene? value)
         writer.WriteStartObject();
 
         writer.WritePropertyName(nameof(Scene.Shapes));
-        ImmutableArraySerializer<IIntersectable>.Instance.Serialize(writer, value.Shapes);
+        ImmutableArraySerializer<IIntersectable>.Instance.Serialize(writer, value.Shapes.ToImmutableArray());
```

Comment:
Would be better to have a separate `IntersectableListSerializer` for this (can be `ProxySerializer`)

Reply:
Will add ticket for that, no need to do it right now